src/backend/cpu/wrap.cpp

```diff
@@ -18,37 +18,34 @@ namespace cpu
 {
 
 template<typename T>
-Array<T> wrap(const Array<T> &in,
-              const dim_t ox, const dim_t oy,
-              const dim_t wx, const dim_t wy,
-              const dim_t sx, const dim_t sy,
-              const dim_t px, const dim_t py,
-              const bool is_column)
+void wrap(Array<T> &out, const Array<T> &in,
+          const dim_t ox, const dim_t oy,
+          const dim_t wx, const dim_t wy,
+          const dim_t sx, const dim_t sy,
+          const dim_t px, const dim_t py,
+          const bool is_column)
 {
-    af::dim4 idims = in.dims();
-    af::dim4 odims(ox, oy, idims[2], idims[3]);
-
-    Array<T> out = createValueArray<T>(odims, scalar<T>(0));
-    out.eval();
     in.eval();
+    const dim4 idims = in.dims();
+    const dim4 odims(ox, oy, idims[2], idims[3]);
+    out = createValueArray<T>(odims, scalar<T>(0));
```

Comment:
I think you meant to omit these lines, correct? Since this will make wrap write its results to the memory returned by `createValueArray` and not the already provided space in the `out` argument. I see that you omitted this in the CUDA and OpenCL backends so I guess it was just an accident that you left this here.